backend/assignments/1/src/main/java/shrishti/example/Coin.java

```diff
@@ -0,0 +1,111 @@
+package shrishti.example;
+import shrishti.example.Logger.logger;
+
+public class Coin {
+    private int rank;
+    private String name;
+    private String symbol;
+    private double price;
+    private long availableQuantity;
+    private double circulatingSupply;
+
+    public Coin() {
+
+    }
+
+
+    public int getRank(){
+        return rank;
+    }
+    public void setRank(int rank){
+        this.rank=rank;
+    }
+    public String getName(){
+        return name;
+    }
+    public void setName(String name){
+        this.name=name;
+    }
+
+    public String getSymbol(){
+        return symbol;
+    }
+    public void setSymbol(String symbol){
+        this.symbol=symbol;
+
+    }
+    public double getPrice(){
+        return price;
+    }
+    public void setPrice(Double price){
+        this.price=price;
+    }
+    public double getCirculatingSupply(){
+        return circulatingSupply;
+    }
+    public void setCirculatingSupply(double circulatingSupply){
+     this.circulatingSupply=circulatingSupply;
+    }
+    public synchronized long getavailableQuantity(){
+        return availableQuantity;
+    }
+
+
+
+    public synchronized void increaseQuantity(long amount){
```

Comment:
What if user1 needs to work on Coin1 and user2 on coin2.
But user2 will still be in wait state untill the method reopoens lock.
Can the locking be in a better place?